src/middleware/pagination_middleware.go

```diff
@@ -0,0 +1,34 @@
+package middleware
+
+import (
+	"github.com/gin-gonic/gin"
+	"go.uber.org/zap"
+	"net/http"
+)
+
+const (
+	PageCtxKey       = "page"
+	pageDefaultValue = "1"
+	LimitCtxKey      = "limit"
+)
+
+// PaginationMiddleware is a middleware for extracting and setting pagination parameters from the request context
+func PaginationMiddleware() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		ctxLogger, exists := c.Get(LoggerCtxKey)
+		if !exists {
+			c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{"error": "logger not set in context"})
+			return
+		}
+		logger := ctxLogger.(*zap.Logger)
+
+		logger.Info("setting up pagination middleware")
+
+		page := c.DefaultQuery(PageCtxKey, pageDefaultValue)
+		limit := c.Query(LimitCtxKey)
```

Comment:
use bindQuery and create type of pagination and then you can ensure the values are int  and set max and min values.